File: challenges/2026-09-16-busker-booking-maximizer/solution.java
import java.util.*;

public class solution {

    public static int maxTotalDonations(int[][] slots) {
        // TODO: implement
        return 0;
    }

    public static void main(String[] args) {
        List<Object[]> tests = new ArrayList<>();
        tests.add(new Object[]{"example 1", new int[][]{{0,3,5}, {2,5,6}, {4,7,5}, {6,9,4}}, 10});
        tests.add(new Object[]{"example 2", new int[][]{{1,4,3}}, 3});
        tests.add(new Object[]{"example 3", new int[][]{{0,2,5}, {0,2,10}, {1,3,3}}, 10});
        tests.add(new Object[]{"edge: back-to-back slots both taken", new int[][]{{0,2,3}, {2,4,4}}, 7});
        tests.add(new Object[]{"edge: single slot", new int[][]{{5,10,100}}, 100});

        for (Object[] test : tests) {
            String name = (String) test[0];
            int[][] slots = (int[][]) test[1];
            int expected = (Integer) test[2];
            int got = maxTotalDonations(slots);
            String status = got == expected ? "PASS" : "FAIL";
            System.out.println("[" + status + "] " + name + ": expected=" + expected + " actual=" + got);
        }
    }
}

File: challenges/2026-09-16-busker-booking-maximizer/solution.go
package main

import "fmt"

func maxTotalDonations(slots [][]int) int {
	// TODO: implement
	return 0
}

type testCase struct {
	name     string
	slots    [][]int
	expected int
}

func main() {
	tests := []testCase{
		{"example 1", [][]int{{0, 3, 5}, {2, 5, 6}, {4, 7, 5}, {6, 9, 4}}, 10},
		{"example 2", [][]int{{1, 4, 3}}, 3},
		{"example 3", [][]int{{0, 2, 5}, {0, 2, 10}, {1, 3, 3}}, 10},
		{"edge: back-to-back slots both taken", [][]int{{0, 2, 3}, {2, 4, 4}}, 7},
		{"edge: single slot", [][]int{{5, 10, 100}}, 100},
	}

	for _, tc := range tests {
		got := maxTotalDonations(tc.slots)
		status := "FAIL"
		if got == tc.expected {
			status = "PASS"
		}
		fmt.Printf("[%s] %s: expected=%d actual=%d\n", status, tc.name, tc.expected, got)
	}
}
